src/storage/compactor/src/server.rs

```diff
@@ -145,7 +145,7 @@ pub async fn prepare_start_parameters(
     ));
 
     let heap_profiler = HeapProfiler::new(
-        total_memory_available_bytes,
+        system_memory_available_bytes(),
```

Comment:
Same reason above, compactor should not use memory more than `(resource_util::memory::system_memory_available_bytes() as f64
            * config.storage.compactor_memory_available_proportion) as usize;` in normal case